src/shells/layer_shell.rs

```diff
@@ -0,0 +1,141 @@
+use bevy::prelude::{Component, Resource};
+use raw_window_handle::{
+    DisplayHandle, HasDisplayHandle, HasWindowHandle, RawDisplayHandle, RawWindowHandle,
+    WaylandDisplayHandle, WaylandWindowHandle, WindowHandle,
+};
+use smithay_client_toolkit::{
+    delegate_layer,
+    globals::GlobalData,
+    reexports::{
+        client::{
+            Connection, Dispatch, Proxy, QueueHandle, globals::GlobalList,
+            protocol::wl_surface::WlSurface,
+        },
+        protocols_wlr::layer_shell::v1::client::{zwlr_layer_shell_v1, zwlr_layer_surface_v1},
+    },
+    shell::{
+        WaylandSurface,
+        wlr_layer::{LayerShell, LayerShellHandler, LayerSurface, LayerSurfaceData},
+    },
+};
+
+pub use smithay_client_toolkit::shell::wlr_layer::{Anchor, KeyboardInteractivity, Layer};
+
+use crate::state::SmithayRunnerState;
+
+#[derive(Debug, Resource, Component, Clone)]
+pub struct LayerShellSettings {
+    pub anchor: Anchor,
+    pub size: (u32, u32),
+    pub exclusive_zone: i32,
+    pub margin: (i32, i32, i32, i32),
+    pub keyboard_interactivity: KeyboardInteractivity,
+    pub layer: Layer,
+}
+
+impl Default for LayerShellSettings {
+    fn default() -> Self {
+        Self {
+            anchor: Anchor::empty(),
+            exclusive_zone: Default::default(),
+            margin: Default::default(),
+            size: (256, 256),
```

Comment:
Should we assume this size or 0, 0? Also is there a provision for dynamically sizing based on the output size in layer shell protocol?